moveit_kinematics/ikfast_kinematics_plugin/templates/ikfast61_moveit_plugin_template.cpp

```diff
@@ -1136,7 +1283,24 @@ bool IKFastKinematicsPlugin::getPositionIK(const geometry_msgs::Pose& ik_pose, c
   }
 
   KDL::Frame frame;
-  tf::poseMsgToKDL(ik_pose, frame);
+  if (transform_tip_ || transform_base_)
+  {
+    geometry_msgs::Pose ik_pose_chain;
+    Eigen::Affine3d ik_eigen_pose_chain;
+    Eigen::Affine3d ik_eigen_pose;
+    tf::poseMsgToEigen(ik_pose, ik_eigen_pose);
+    if (!transformToChainFrame(ik_eigen_pose, ik_eigen_pose_chain))
+    {
+      ROS_ERROR_NAMED("ikfast", "failed to transform input pose to ik chain frame");
+      return false;
+    }
+    tf::poseEigenToMsg(ik_eigen_pose_chain, ik_pose_chain);
+    tf::poseMsgToKDL(ik_pose_chain, frame);
+  }
+  else
+  {
+    tf::poseMsgToKDL(ik_pose, frame);
+  }
```

Comment:
this code section looks the same as line 1087 above - i think this should be a helper function

Reply:
done